index.js

```diff
@@ -51,6 +53,71 @@ app.post("/audio_info", upload.single("file"), async (req, res) => {
   }
 });
 
+app.post("/spotify-auth", async (req, res) => {
+  // console.log("request.body", req.body);
+  const payload = {
+    grant_type: "authorization_code",
+    code: req.body.code,
+    redirect_uri: process.env.REDIRECT_URI,
+  };
+  try {
+    const result = await axios({
+      method: "post",
+      url: "https://accounts.spotify.com/api/token",
+      data: qs.stringify(payload),
+      headers: {
+        Authorization: `Basic ${process.env.CLIENT_DETAILS}`,
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    });
+    res.send(result.data);
+    console.log("result.data", result.data);
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+
+
+app.get(`/artist/:id/top-tracks`, async (req, res) => {
+  const id = req.params.id;
+ try {
+  
+  const result = await axios({
+    method: "get",
+    url: `https://api.spotify.com/v1/artists/${id}/top-tracks?market=GB`,
+    headers: {
+      Authorization: `Bearer ${process.env.SPOTIFY_BEARER_TOKEN}`,
+      'Content-Type': 'application/json'
+    }
+
+  })
+  console.log(result.data.tracks[1], 'result')
+  res.send(result.data);
+ } catch (error) {
+  console.log(error)
+ }
+})
+
+app.get(`/artist/:id/related-artists`, async (req, res) => {
+  const id = req.params.id;
+ try {
+  
+  const result = await axios({
+    method: "get",
+    url: `https://api.spotify.com/v1/artists/${id}/related-artists`,
+    headers: {
+      Authorization: `Bearer ${process.env.SPOTIFY_BEARER_TOKEN}`,
```

Comment:
again should this be passed as a header not a ENV?

Reply:
Same as above